implementations/rs/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+use crate::wrap::wrap_info::get_manifest;
+use connection::WalletError;
+use connections::Connections;
+use ethers::{
+    prelude::SignerMiddleware,
+    providers::{Http, Middleware, PendingTransaction, Provider},
+    signers::Signer,
+    types::{
+        transaction::{eip2718::TypedTransaction, eip712::TypedData},
+        TransactionRequest, TxHash,
+    },
+};
+use handler::{
+    handle_call_params, handle_get_block_by_number_params, handle_send_params,
+    handle_sign_typed_data_args,
+};
+use polywrap_core::invoker::Invoker;
+use polywrap_plugin::{
+    error::PluginError,
+    implementor::plugin_impl,
+    JSON::{self, from_str, from_value, to_string, to_value, Value},
+};
+use std::{str::FromStr, sync::Arc};
+use tokio::runtime::Runtime;
+use wrap::module::{
+    ArgsRequest, ArgsSignMessage, ArgsSignTransaction, ArgsSignerAddress, ArgsWaitForTransaction,
+    Module,
+};
+
+pub mod connection;
+pub mod connections;
+mod handler;
+mod networks;
+mod wrap;
+
+#[derive(Debug)]
+pub struct EthereumWalletPlugin {
+    connections: Connections,
+}
+
+impl EthereumWalletPlugin {
+    pub fn new(connections: Connections) -> Self {
+        Self { connections }
+    }
+
+    fn parse_parameters(&self, method: &str, params: &Option<String>) -> Vec<Value> {
```

Comment:
Both of these helper methods can be static functions exported from another module.